src/rill.h

```diff
@@ -186,7 +197,9 @@ struct rill_pairs *rill_query_vals(
         const rill_val_t *vals, size_t len,
         struct rill_pairs *out);
 
-struct rill_pairs *rill_query_all(const struct rill_query *query);
+struct rill_pairs *rill_query_all(
```

Comment:
Not sure if it's still a WIP so take with a grain of salt.

If we're going with this style of interface (which I think we should, I was just going to do it later) then I would remove the other `rill_query_xxx` function and just have one `rill_query` function.